### PR TITLE
[FIX] web, google_drive: fixed display of field link

### PR DESCRIPTION
**Pull request by kebeclibre**
Before this commit, a field handled with a widget url was given its value (href) as its text, fully displaying the url.
Moreover, the display was odd and did not match that of buttons

After this commit, if the field contains a text attribute, we use is as the text of the link.
The link also correctly displays and looks like a button (in a form view)

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/google_drive/views/res_config_settings_views.xml [hunk 1]
```diff
@@ -13,9 +13,9 @@
                         <field name="google_drive_authorization_code"/>
                     </div>
                     <div>
-                        <a type="button" class="oe-link" target="_blank"><i class="fa fa-fw fa-arrow-right"/>
-                            <field name="google_drive_uri" widget="url" text=" Generate Code"/> 
-                        </a>
+                        <span type="button" class="btn-link btn-link-as-button fa-arrow-right">
+                            <field name="google_drive_uri" widget="url" text="Get Authorization Code"/>
```
Comment (mge-odoo) [inline addons/google_drive/views/res_config_settings_views.xml]: No need to change .po/.pot when you change this?

File: addons/web/static/src/js/fields/basic_fields.js [hunk 2]
```diff
@@ -1006,7 +1006,7 @@ var UrlWidget = InputField.extend({
      * @private
      */
     _renderReadonly: function () {
-        this.$el.text(this.value)
+        this.$el.text(this.attrs.text || this.value)
```
Comment (mge-odoo) [inline addons/web/static/src/js/fields/basic_fields.js]: That's very strange ; why should a field widget take as value something else than its value?
Why using a field widget then, and not just a static link?

File: addons/web/static/src/less/views.less [hunk 3]
```diff
@@ -44,3 +44,13 @@
         background-color: @brand-danger;
     }
 }
+
+.btn-link-as-button {
```
Comment (mge-odoo) [inline addons/web/static/src/less/views.less]: All odoo specific classes start with "o_" ; here you try to trick devs to make them believe that your class is a bootstrap-like class.

**Comment by kebeclibre** [comment]
closed at e64762e535504a82c5a55615b9a3e17dbf7ac562

**Pull request closed without merging**